### PR TITLE
Implemented the JWT Authentication and Authorization,

### PR DESCRIPTION
**Pull request by Venkadesan0210**
1)Created the Controller to access the endpoint.
2)Created the Dto to access the Request for internal purpose. 
3)Created the Two Repo for mapping the DataBase.
4)Created the service class and interface to implement the methods.

### CODE REVIEW

File: LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/dto/RoleDto.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.example.LeaveManagement.dto;
+
+public class RoleDto {
+    private String roleName;
+    private String roleDescription;
+
```
Comment (shyamsundarv1489) [inline LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/dto/RoleDto.java]: Remove empty lines

File: LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        Set<Role> userRoles = new HashSet<>();
+        userRoles.add(role);
+        user.setUserName(userDTO.getUserName());
+        user.setRole(userRoles);
+        user.setUserPassword(getEncodedPassword(userDTO.getUserPassword()));
+        user.setUserRole(role.getRoleName());
+        user.setUserFirstName(userDTO.getUserFirstName());
+        user.setUserLastName(userDTO.getUserLastName());
+
+
```
Comment (shyamsundarv1489) [inline LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/service/UserService.java]: Remove empty lines

File: LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/service/interfaces/UserServiceInt.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.example.LeaveManagement.service.interfaces;
+import com.example.LeaveManagement.dto.UserDTO;
+import com.example.LeaveManagement.entity.User;
+
+import java.util.List;
+
+public interface UserServiceInt {
```
Comment (shyamsundarv1489) [inline LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/service/interfaces/UserServiceInt.java:5]: Try to give meaningful names.

File: LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/configuration/WebSecurityConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    @Override
+    public AuthenticationManager authenticationManagerBean() throws Exception {
+        return super.authenticationManagerBean();
+    }
+
+    @Override
+    protected void configure(HttpSecurity httpSecurity) throws Exception {
+        httpSecurity.cors();
+        httpSecurity.csrf().disable()
+                .authorizeRequests().antMatchers("/authenticate", "/registerNewUser", "/createNewRole", "/allusers","/update-user","/leaves/**","/apply-leave","/update-leave","/leaves-manager").permitAll()
```
Comment (shyamsundarv1489) [inline LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/configuration/WebSecurityConfiguration.java:45]: As discussed ensure this is updated to handle requests based on roles.

File: LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/configuration/JwtRequestFilter.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+        String username = null;
+        String jwtToken = null;
+
+        if (requestTokenHeader != null && requestTokenHeader.startsWith("Bearer ")) {
+            jwtToken = requestTokenHeader.substring(7);
+            try {
+                username = jwtUtil.getUsernameFromToken(jwtToken);
+            } catch (IllegalArgumentException e) {
+                logger.error("Unable to get JWT Token", e);
```
Comment (shyamsundarv1489) [inline LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/configuration/JwtRequestFilter.java:40]: Log the error message properly

File: LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/dto/RoleDto.java (hunk 1 above)
Comment (Venkadesan0210) [reply LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/dto/RoleDto.java]: Addressed the above comments

File: LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/service/UserService.java (hunk 2 above)
Comment (Venkadesan0210) [reply LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/service/UserService.java]: Addressed the above comments

File: LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/service/interfaces/UserServiceInt.java (hunk 3 above)
Comment (Venkadesan0210) [reply LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/service/interfaces/UserServiceInt.java:5]: Addressed the above comments

File: LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/configuration/WebSecurityConfiguration.java (hunk 4 above)
Comment (Venkadesan0210) [reply LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/configuration/WebSecurityConfiguration.java:45]: Yes, This method can handle the API endpoints based on role. However, for testing purposes, I have granted access to all endpoints for all roles.

File: LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/configuration/JwtRequestFilter.java (hunk 5 above)
Comment (Venkadesan0210) [reply LeaveManagement/LeaveManagement/src/main/java/com/example/LeaveManagement/configuration/JwtRequestFilter.java:40]: Addressed the above comments

**Pull request merged**